sotopia/database/evaluation_dimensions.py

```diff
@@ -0,0 +1,232 @@
+from redis_om import JsonModel
+from redis_om.model.model import Field
+from pydantic import BaseModel, create_model
+from typing import Type, Callable, Tuple, Annotated, Union
+
+
+class CustomEvaluationDimension(JsonModel):
+    name: str = Field(index=True)
+    description: str = Field(index=True)
+    range_high: int = Field(index=True)
+    range_low: int = Field(index=True)
+
+
+class CustomEvaluationDimensionList(JsonModel):
+    name: str = Field(index=True)
+    dimension_ids: list[str] = Field(default_factory=lambda: [], index=True)
+
+
+class EvaluationDimensionGenerator:
+    @staticmethod
+    def create_range_validator(low: int, high: int) -> Callable[[int], bool]:
+        def validator(x: Tuple[str, int]) -> Tuple[str, int]:
+            if not isinstance(x, tuple) or len(x) != 2:
+                raise ValueError("Must be a tuple of (str, int)")
+            if not isinstance(x[1], int) or not low <= x[1] <= high:
+                raise ValueError(f"Score must be between {low} and {high}")
+            return x
+
+        return validator
+
+    @staticmethod
+    def generate_dimension_model(dimension_ids: list[str]) -> Type[BaseModel]:
```

Comment:
what is this function used for?

Reply:
create an validator for the evaluation metric?